dell/xps/13-9310/default.nix

```diff
@@ -0,0 +1,89 @@
+{ lib, pkgs, ... }:
+{
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+  ];
+
+  # TODO: upstream to NixOS/nixpkgs
+  nixpkgs.overlays = [(final: previous: {
+    qca6390-firmware = final.callPackage ./qca6390-firmware.nix {};
+  })];
+
+  hardware.firmware = lib.mkBefore [
+    # Necessary for audio.
+    # https://bbs.archlinux.org/viewtopic.php?pid=1933548#p1933548
+    pkgs.sof-firmware
```

Comment:
On my system (20.09) `sof-firmware` was already in this set.

I don't suppose you've had success with the microphone yet?